script.js

```diff
@@ -0,0 +1,59 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const bookTitle = document.getElementById('title');
+  const bookAuthor = document.getElementById('author');
+  const article = document.getElementById('bookList');
+
+  const addButton = document.getElementById('add');
+  const saveItemToLocalStorage = (title, author) => {
+    const savedBooks = JSON.parse(localStorage.getItem('savedBooks')) || [];
+    savedBooks.push({ title, author });
+    localStorage.setItem('savedBooks', JSON.stringify(savedBooks));
+  };
+
+  const removeItemFromLocalStorage = (div) => {
+    const savedBooks = JSON.parse(localStorage.getItem('savedBooks')) || [];
+    const title = div.querySelector('p:first-child').innerHTML;
+    const author = div.querySelector('p:nth-child(2)').innerHTML;
+    const index = savedBooks.findIndex(
+      (item) => item.title === title && item.author === author,
+    );
+    if (index !== -1) {
+      savedBooks.splice(index, 1);
```

Comment:
Well done on your implementation! However, I'd like to suggest a small change that could enhance your code quality.

- [ ] [Optional]  Currently, you are using the `array.splice()` method to remove elements from the array. While this approach works as intended, I recommend considering the use of the `array.slice()` method instead. The [slice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) method of [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) instances returns a [shallow copy](https://developer.mozilla.org/en-US/docs/Glossary/Shallow_copy) of a portion of an array, selected from the start index to the end index (end index not included). This new array copy includes the items within the specified range, leaving the original array untouched. The key advantage of using `array.slice()` for this purpose is that it maintains the immutability of the original array. This means that the removal of a book using `array.slice()` won't modify the existing array but will return a new array with the desired items removed. This aligns well with the principles of data immutability and can lead to more predictable and maintainable code.

In summary, while both `array.splice()` and `array.slice()` are valid methods for removing elements, the choice to use `array.slice()` can contribute to clearer code that adheres to the principles of immutability. Keep up the excellent work, and happy coding! 🌟📝🚀